tests/e2e-tests/specs/front-end/front-end-checkout.test.js

```diff
@@ -0,0 +1,151 @@
+		// Sell to all countries
+		await expect( page ).toSelect( '#woocommerce_allowed_countries', 'Sell to all countries' );
+		// Set currency to USD
+		await expect( page ).toSelect( '#woocommerce_currency', 'United States (US) dollar ($)' );
+		// Tax calculation should have been enabled by another test - no-op
+		// Save
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await Promise.all( [
+			expect( page ).toMatchElement( '#message', { text: 'Your settings have been saved.' } ),
+			expect( page ).toMatchElement( 'select[name="woocommerce_default_country"]', { text: 'United States (US) — California' } ),
+			expect( page ).toMatchElement( '#woocommerce_allowed_countries', { text: 'Sell to all countries' } ),
+			expect( page ).toMatchElement( '#woocommerce_currency', { text: 'United States (US) dollar ($)' } ),
+		] );
+
+		// Enable BACS payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'bacs' );
+		await setCheckbox( '#woocommerce_bacs_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_bacs_enabled' );
+
+		// Enable COD payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'cod' );
+		await setCheckbox( '#woocommerce_cod_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_cod_enabled' );
+
+		// Enable PayPal payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'paypal' );
+		await setCheckbox( '#woocommerce_paypal_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_paypal_enabled' );
+
+		await StoreOwnerFlow.logout();
+	} );
+
+	it( 'should displays cart items in order review', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+
+		await CustomerFlow.goToCheckout();
+		await CustomerFlow.productIsInCheckout( 'Simple product', 1, 9.99 );
+		await expect( page ).toMatchElement( '.cart-subtotal .amount', { text: '$9.99' } );
+	} );
+
+	it( 'allows customer to choose available payment methods', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+
+		await expect( page ).toClick( '.wc_payment_method label', { text: 'PayPal' } );
+		await expect( page ).toClick( '.wc_payment_method label', { text: 'Direct bank transfer' } );
+		await expect( page ).toClick( '.wc_payment_method label', { text: 'Cash on delivery' } );
+	} );
+
+	it( 'allows customer to fill billing details', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+
+		await expect( page ).toFill( '#billing_first_name', 'John' );
```

Comment:
Should we abstract the code to fill the address into its own function to avoid repeating it below?

Reply:
Good catch, thanks! I addressed that by creating appropriate functions for filling out billing and shipping addresses.